server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala

```diff
@@ -1410,6 +1414,165 @@ class DeltaSharingServiceSuite extends FunSuite with BeforeAndAfterAll {
+      "col-0d00ce77-f27b-4778-b0b0-ca4b8289ca21" -> "2024-06-25").toMap
+    val expectedFiles = Seq(
+      DeltaResponseFileAction(
+        id = "4b90398804797110234e987667aeef902658055cfe3be02281d1ce1d4fe762ca",
+        version = null,
+        timestamp = null,
+        expirationTimestamp = actualFiles(0).expirationTimestamp,
+        deltaSingleAction = DeltaSingleAction(
+          add = DeltaAddFile(
+            path = actualFiles(0).deltaSingleAction.add.path,
+            partitionValues = firstExpectedPartitionMap,
+            size = 653,
+            modificationTime = 1719346426000L,
+            dataChange = true,
+            stats = """{"numRecords":1,"minValues":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":"2024-06-23T00:00:00.000Z"},"maxValues":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":"2024-06-23T00:00:00.000Z"},"nullCount":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":0}}"""
+          )
+        )
+      ), DeltaResponseFileAction(
+        id = "c351b3936ee9230044de858bfe7262b757e68496d57b8f2be61db55997141cb0",
+        version = null,
+        timestamp = null,
+        expirationTimestamp = actualFiles(1).expirationTimestamp,
+        deltaSingleAction = DeltaSingleAction(
+          add = DeltaAddFile(
+            path = actualFiles(1).deltaSingleAction.add.path,
+            partitionValues = secondExpectedPartitionMap,
+            size = 652,
+            modificationTime = 1719346426000L,
+            dataChange = true,
+            stats = """{"numRecords":1,"minValues":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":"2024-06-24T01:00:00.000Z"},"maxValues":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":"2024-06-24T01:00:00.000Z"},"nullCount":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":0}}"""
+          )
+        )
+      ), DeltaResponseFileAction(
+        id = "016f8b0bd53cf812e1fb198b89ddd8f9136136fe676936cb1b9159939d401e15",
+        version = null,
+        timestamp = null,
+        expirationTimestamp = actualFiles(2).expirationTimestamp,
+        deltaSingleAction = DeltaSingleAction(
+          add = DeltaAddFile(
+            path = actualFiles(2).deltaSingleAction.add.path,
+            partitionValues = thirdExpectedPartitionMap,
+            size = 653,
+            modificationTime = 1719346426000L,
+            dataChange = true,
+            stats = """{"numRecords":1,"minValues":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":"2024-06-25T00:00:00.000Z"},"maxValues":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":"2024-06-25T00:00:00.000Z"},"nullCount":{"col-6c51f03a-5218-4630-9c68-fe5f86060101":0}}"""
+          )
+        )
+      )
+    )
+    assert(expectedFiles == actualFiles.toList)
+    assert(actualFiles.count(_.expirationTimestamp != null) == 3)
+    verifyPreSignedUrl(actualFiles(0).deltaSingleAction.add.path, 653)
+    verifyPreSignedUrl(actualFiles(1).deltaSingleAction.add.path, 652)
+    verifyPreSignedUrl(actualFiles(2).deltaSingleAction.add.path, 653)
+  }
+
```

Comment:
And add two tests for `/metadata` on the two tables.

Reply:
@linzhou-db is this necessary since I am checking the metadata and protocol in the query function anyways? If I do write a test just for query it will just be the top half of this test copied and pasted directly

Reply:
Yes this is necessary, as we need to make sure the rpc works as expected. 

And actually, we want to add test with different header values for `deltaSharingCapabilities` and verify the server behavior.

Reply:
Ok I will add the 2 metadata tests

For the other part, should I pass nothing into the readerFeatures (as opposed to passing in readerFeatures = columnMapping) and see if an error is being thrown (it should throw an error in assertTableProperties and this would address the other comment you made right)?

Reply:
For the other part, we should test:
- responseFormat=parquet;readerFeatures=columnMapping -> this should result in error
- responseFormat=parquet,delta;readerFeatures=columnMapping -> this should result in delta format response
- responseFormat=deltal; -> with empty reader features, which should throw error. BUT, it may not throw exception as we expected, and we may want to skip it for now and double check why the reader features is empty in the tables I created for testing.

Reply:
added tests for these 3 and it all worked as expected